launcher/minecraft/MinecraftInstance.cpp

```diff
@@ -438,15 +440,15 @@ QStringList MinecraftInstance::javaArguments()
 
     args << "-Duser.language=en";
 
+    if (javaVersion.isModular() && traits().contains("legacySkins") && settings()->get("LegacySkinFix").toBool())
+        args << "--add-opens" << "java.base/java.net=ALL-UNNAMED";
```

Comment:
q: could you explain the purpose of this argument?

Reply:
Without this an error appears on newer Java versions due to using reflection.

Reply:
Could this not be enabled regardless of the fix then? It sounds like it'd be useful in other cases.
I also think it should be commented to provide context.

Reply:
I agree with adding a comment, but I don't think using this argument is considered good practice. It should only automatically be added if it's necessary, and it also only applies to java.net.